edb/schema/functions.py

```diff
@@ -1879,6 +1879,19 @@ def _create_begin(
                         f'{p_type.get_displayname(schema)}',
                         context=self.source_context)
 
+        # Make sure variadic parameters do not contain optional types in
+        # user-defined functions
+        if language == qlast.Language.EdgeQL:
+            if variadic := params.find_variadic(schema):
+                typemod = variadic.get_typemod(schema)
+                if typemod is ft.TypeModifier.OptionalType:
+                    raise errors.InvalidFunctionDefinitionError(
+                        f'cannot create the `{signature}` function: '
+                        f'variadic argument '
+                        f'`{variadic.get_displayname(schema)}` '
+                        f'illegally declared with optional type in user code',
```

Comment:
I'd drop the `in user code` part; users will only be defining edgeql user functions

Reply:
Discussed. Changed to "in user-defined function" to match many other errors in the file.